lib/ts/recipe/thirdparty/recipeImplementation.ts

```diff
@@ -0,0 +1,262 @@
+                      state,
+                      redirect_uri: input.thirdPartyRedirectionURL,
+                  });
+
+            return {
+                status: "OK",
+                url: urlWithState,
+            };
+        },
+
+        getOAuthAuthorisationURL: async function (input: {
+            thirdPartyProviderId: string;
+            config: NormalisedInputType;
+            userContext: any;
+            options?: RecipeFunctionOptions;
+        }): Promise<{
+            status: "OK";
+            url: string;
+            fetchResponse: Response;
+        }> {
+            const { jsonBody, fetchResponse } = await querier.get<{
+                status: "OK";
+                url: string;
+            }>(
+                "/authorisationurl",
+                {},
+                { thirdPartyId: input.thirdPartyProviderId },
+                Querier.preparePreAPIHook({
+                    config: input.config,
+                    action: "GET_AUTHORISATION_URL",
+                    options: input.options,
+                    userContext: input.userContext,
+                }),
+                Querier.preparePostAPIHook({
+                    config: input.config,
+                    action: "GET_AUTHORISATION_URL",
+                    userContext: input.userContext,
+                })
+            );
+
+            return {
+                status: "OK",
+                url: jsonBody.url,
+                fetchResponse,
+            };
+        },
+
+        signInAndUp: async function (input: {
+            thirdPartyProviderId: string;
+            thirdPartyProviderClientId?: string;
+            thirdPartyRedirectionURL: string;
+            config: NormalisedInputType;
+            userContext: any;
+            options?: RecipeFunctionOptions;
+        }): Promise<
+            | {
+                  status: "OK";
+                  user: UserType;
+                  createdNewUser: boolean;
+                  fetchResponse: Response;
+              }
+            | {
+                  status: "NO_EMAIL_GIVEN_BY_PROVIDER";
+                  fetchResponse: Response;
+              }
+            | {
+                  status: "FIELD_ERROR";
+                  error: string;
+                  fetchResponse: Response;
+              }
+        > {
+            const stateFromStorage = this.getOAuthState({
+                userContext: input.userContext,
+                config: input.config,
+            }).state;
+
+            const code = getQueryParams("code");
+
+            const stateFromQueryParams = getQueryParams("state");
+
+            if (
+                getQueryParams("error") !== null ||
+                stateFromStorage === undefined ||
+                stateFromStorage.thirdPartyId !== input.thirdPartyProviderId ||
+                stateFromStorage.state !== stateFromQueryParams ||
+                code === null
+            ) {
+                throw new STGeneralError("");
```

Comment:
Throw a regular error here since we want to show a generic swr message

Reply:
Done